dali/kernels/imgproc/geom/remap_npp.h

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_GEOM_REMAP_NPP_H_
+#define DALI_KERNELS_IMGPROC_GEOM_REMAP_NPP_H_
+
+#include <nppi_geometry_transforms.h>
+#include <tuple>
+#include "dali/core/geom/box.h"
+#include "dali/core/span.h"
+#include "dali/kernels/imgproc/geom/npp_remap_call.h"
+#include "dali/kernels/imgproc/geom/remap.h"
+#include "include/dali/core/backend_tags.h"
+#include "include/dali/core/tensor_view.h"
+
+namespace dali::kernels::remap {
+
+namespace detail {
+
+/**
+ * Return a default ROI. Default ROI is a whole image.
+ */
+template<int ndims>
+Box<2, int64_t> default_roi(const TensorShape<ndims> &ts) {
+  assert(ts.sample_dim() >= 2);
+  return {{0,     0},
+          {ts[1], ts[0]}};
+}
+
+}  // namespace detail
+
+/**
+ * RemapKernel implementation using NPP.
+ *
+ * @remark NPP doesn't offer Border policy. When using this implementation,
+ *         the border is always BoundaryType::CONSTANT and equal to 0.
+ *
+ * @see RemapKernel.
+ *
+ * @tparam Backend Storage backend, must be GPU-accessible.
+ * @tparam T Type of the input and output data.
+ */
+template<typename Backend, typename T>
+struct NppRemapKernel : public RemapKernel<Backend, T> {
+  using Border = typename RemapKernel<Backend, T>::Border;
+  using MapType = typename RemapKernel<Backend, T>::MapType;
+  using SupportedInputTypes = std::tuple<uint8_t, int16_t, uint16_t, float>;
+  static_assert(has_type_v<T, SupportedInputTypes>, "Unsupported input type.");
+
+
+  void Run(KernelContext &context,
+           TensorListView<Backend, T> output,
+           TensorListView<Backend, const T> input,
+           TensorListView<Backend, const MapType, 2> mapsx,
+           TensorListView<Backend, const MapType, 2> mapsy,
+           span<Box<2, int64_t>> output_rois = {},
+           span<Box<2, int64_t>> input_rois = {},
+           span<DALIInterpType> interpolations = {},
+           span<Border> borders = {}) override {
```

Comment:
Doesn't this function lack some input validation? I guess at least to check `mapsx`, `mapsy` shapes vs. `rois`?

Reply:
Done